Flank/src/main/java/com/walmart/otto/filter/TestFilters.java

```diff
@@ -0,0 +1,118 @@
+package com.walmart.otto.filter;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class TestFilters {
+
+  private static final Pattern FILTER_ARGUMENT = Pattern.compile("(.+) (.+)");
+
+  public static TestFilter fromCommandLineArguments(String arguments) {
+    String[] args = arguments.split(";");
+
+    List<TestFilter> filters =
+        Arrays.stream(args)
+            .map(String::trim)
+            .map(TestFilters::parseSingleFilter)
+            .collect(Collectors.toList());
+
+    Collections.addAll(filters, notIgnored());
+
+    return allOf(filters);
+  }
+
+  public static TestFilter createDefault() {
+    return notIgnored();
+  }
+
+  private static TestFilter parseSingleFilter(String argument) {
+    Matcher matcher = FILTER_ARGUMENT.matcher(argument);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid argument: " + argument);
+    }
+
+    Collection<String> args =
+        Arrays.stream(matcher.group(2).split(",")).collect(Collectors.toSet());
```

Comment:
I am kind of confused with the API

The main regex is `X Y` where X is the command, how can Y contain spaces?
Doesn't `package com.p1, com.p2; class x` break the regex?

Instead of having `,` and `;` I would prefer having only `,`
User will need to write the `canonicalCommand` all the time, example:
`package com.p1, package com.p2, class x`

wdyt?

Reply:
> Instead of having , and ; I would prefer having only ,
User will need to write the canonicalCommand all the time, example:
package com.p1, package com.p2, class x

I think is a good point. I took inspiration from the [AndroidJUnitRunner](https://developer.android.com/reference/android/support/test/runner/AndroidJUnitRunner.html) interface but I think we are free to diverge and writing the canonical command all the time is cleaner. I'll change that. Thanks for the suggestion!

Reply:
After a second though, I think I'll leave it as it is because such a change would affect the semantic:

- `package com.p1, package com.p2, annotation x` means "run all the tests in package com.p1 AND package com.p2 AND annotation x" 
- `package com.p1, com.p2; annotation x` means "run all the tests in package com.p1 OR package com.p2 AND annotation x`.

Reply:
Oh, I didn't know you were basing your implementation on `AndroidJUnitRunner`. I know see their API. Why didn't you just copy their implementation?

Reply:
It would have probably taken more time to understand and adapt their code than actually writing this 100 lines of code.